packages/dds/cell/src/cell.ts

```diff
@@ -158,23 +225,44 @@ export class SharedCell<T = any> extends SharedObject<ISharedCellEvents<T>> impl
         return this.data === undefined;
     }
 
+    /**
+     * {@inheritDoc ISharedCell.getAttribution}
+     */
+    public getAttribution(): AttributionKey | undefined {
+        return this.attribution;
+    }
+
+    /**
+     * {@inheritDoc ISharedCell.setAttribution}
+     */
+    public setAttribution(message: ISequencedDocumentMessage): void {
+        this.attribution = { type: "op", seq: message.sequenceNumber };
+    }
+
+    /**
+     * {@inheritDoc ISharedCell.hastAttribution}
+     */
+    public hasAttribution(): boolean {
+        return this.attribution !== undefined;
+    }
+
     /**
      * Creates a summary for the Cell.
      *
      * @returns The summary of the current state of the Cell.
      */
     protected summarizeCore(serializer: IFluidSerializer): ISummaryTreeWithStats {
-        const content: ICellValue = { value: this.data };
+        const content: ICellContent = { value: this.data, attribution: this.attribution };
         return createSingleBlobSummary(snapshotFileName, serializer.stringify(content, this.handle));
     }
 
     /**
      * {@inheritDoc @fluidframework/shared-object-base#SharedObject.loadCore}
      */
     protected async loadCore(storage: IChannelStorageService): Promise<void> {
-        const content = await readAndParse<ICellValue>(storage, snapshotFileName);
+        const content = await readAndParse<ICellContent>(storage, snapshotFileName);
 
-        this.data = this.decode(content);
+        this.data = this.decode({ value: content.value });
```

Comment:
looks like this is dropping the attribution data on summary load

Reply:
I checked the functionality of `decode/encode`, it is supposed to be only related to the cell value. So I just add logic about attribution data in `summary/load` and it should provide enough coverage